store/store.go

```diff
@@ -531,11 +531,57 @@ func (s *SnapUbuntuStoreRepository) FindSnaps(searchTerm string, channel string,
 	return snaps, nil
 }
 
-// Updates returns the available updates for a list of snap identified by fullname with channel.
-func (s *SnapUbuntuStoreRepository) Updates(installed []string, auther Authenticator) (snaps []*snap.Info, err error) {
-	// XXX: uses obsolete end point!
+// RefreshCandidate contains information for the store about the currently
+// installed snap so that the store can decide what update we should see
+type RefreshCandidate struct {
+	SnapID   string
+	Revision snap.Revision
+	Epoch    string
+	DevMode  bool
 
-	jsonData, err := json.Marshal(map[string][]string{"name": installed})
+	// the desired channel
+	Channel string
+}
+
+// the exact bits that we need to send to the store
+type currentSnapJson struct {
+	SnapID   string `json:"snap_id"`
+	Channel  string `json:"channel"`
+	Revision int    `json:"revision,omitempty"`
+	Epoch    string `json:"epoch"`
+	DevMode  bool   `json:"devmode"`
```

Comment:
@matiasb So far we've been attempting to have the confinement setting reflecting one single concept: what is the preferred mode of confinement operation by the snap, which is related to but not the same as the current state of the snap. You are right that this confinement setting can have three or more values, but at the values we considered possible so far, such as "enforced", would mean we want the `devmode` to necessarily be false for the current state, rather than being completely orthogonal to it.

There's a small chance we'll have indeed a third mode which is completely orthogonal, but this will break more concepts than simply this API. The `devmode` term is used throughout snapd, including CLI, API, and internal state.